### ISSUE TITLE
algorithms: TimestepControl/ThetaTimestepping

### ISSUE THREAD

**Issue by tjhei** [dealii/dealii#2004]
We have a class ```ThetaTimestepping``` and ```TimestepControl``` in algorithms/ and @Rombur added ```TimeStepping``` in base/time_stepping.h. Can we either deprecate or integrate the two? Any thoughts?

**Comment by kronbichler** [comment]
base/time_stepping.h certainly has the more modern interface. But I'm not sure that @Rombur's approach covers the full functionality of `ThetaTimestepping`. I have not actively used any of the two (I am guilty of re-implementing things like time stepping over and over) but should at some point. Maybe the next time I start a new application...

**Comment by kronbichler** [comment]
I'll let the authors decide whether there is something to be unified. I should add that I like the `algorithms/` subfolder concept for things like time stepping.

**Comment by drwells** [comment]
We haven't made any progress on this in four years :crying_cat_face:

**Comment by kronbichler** [comment]
Given the work on `DiscreteTime`, #8824, I wonder if we should do more here? If yes, it probably won't happen for this release.

**Comment by rezarastak** [comment]
I am planning to use `DiscreteTime` within the implementations of many algorithms that use time-stepping. `DiscreteTime` is designed to be algorithm-agnostic and can be used in many situations. The transformation of the implementations might take a some time but hopefully it can be done before 9.3 release.

**Comment by bangerth** [comment]
I tend to think that we should abandon our own efforts to provide time steppers and instead build on packages such as ARKODE from SUNDIALS. We will likely never be able to do things as well as those guys.

**Comment by blaisb** [comment]
> I tend to think that we should abandon our own efforts to provide time steppers and instead build on packages such as ARKODE from SUNDIALS. We will likely never be able to do things as well as those guys.

That's a good reflexion to have. I think we would require more tutorials were we use these libraries with common deal.II features (Trilinos/PETSC wrappers, non-linear problems, etc.) and it would facilitate their usage. At firsthand, these libraries are insanely powerful, but the learning curve is definitely something. I think that it is a bit similar to how SACADO is difficult to use and how the current work of Jean-Paul Pelteret is insanely useful because it brings out these techniques to a wider audience. Some of the new users of deal.II already struggle with Object oriented concepts (templates, inheritance, etc.). Sometimes we look at these things with our own "expert goggles" (in your case, i'm no expert yet :) ) and we forget how many roadblocks there are to a more common  usage of these tools. I feel like as long as there is no step or example fully using a notion in deal.II, it hinders it's widespread adoption :).

**Comment by bangerth** [comment]
@blaisb Absolutely, and I'm quite mindful about that as well -- that's part of why I wrote step-77 (#11953) to explain how one would approach using SUNDIALS/KINSOL for a nonlinear problem. I would very much like to think that writing tutorials for SUNDIALS/ARKODE would be equally useful.

**Comment by bangerth** [comment]
The time steppers in `namespace Algorithms` are "early deprecated" in #19848. If that patch is merged, that's what will be in 9.8, and these classes will be "really deprecated" in 9.9, and removed after the 9.9 release.

**Comment by drwells** [comment]
Since the classes are now deprecated we can close this.
